packages/tools/src/eventDispatchers/shared/getActiveToolForTouchEvent.ts

```diff
@@ -0,0 +1,55 @@
+import { ToolGroupManager } from '../../store';
+import { MouseBindings, ToolModes } from '../../enums';
+import { EventTypes } from '../../types';
+import getMouseModifier from './getMouseModifier';
+
+const { Active } = ToolModes;
+
+/**
+ * Iterate tool group tools until we find a tool that has a "ToolBinding"
+ * that matches our TouchEvent's `buttons`. It's possible there will be no match
+ * (no active tool for that touch button combination).
+ *
+ * @param evt - The event dispatcher touch event.
+ *
+ * @returns tool
+ */
+export default function getActiveToolForTouchEvent(
+  evt: EventTypes.NormalizedTouchEventType
+) {
+  // Todo: we should refactor this to use getToolsWithModesForTouchEvent instead
+  const { renderingEngineId, viewportId } = evt.detail;
+  const touchEvent = evt.detail.event;
+
+  const toolGroup = ToolGroupManager.getToolGroupForViewport(
+    viewportId,
+    renderingEngineId
+  );
+
+  const toolGroupToolNames = Object.keys(toolGroup.toolOptions);
+
+  const numTouchPoints = Object.keys(touchEvent.touches).length;
+
+  // If any keyboard modifier key is also pressed
+  const modifierKey = getMouseModifier(touchEvent);
```

Comment:
The comment above this line says if `keyboard modifier` is pressed, but the function is `getMouseModifier` is this correct?

Reply:
yes the mousemodifer function is general keyboard related. I think we may need to change the function name but this PR is getting pretty large + there are other tool related PRs coming.